Address form

chryslovelace opened the pull request:
When the user selects 'Home Address' from the request location dropdown, shows their home address under the request box, with a button that opens a form allowing them to type in a temporary address. After doing that, it shows that address, and buttons to edit it again, or revert to their home address.

alex-wreschnig committed:
Rick's first cut of modularizing the thirdIron integration

wopsononock committed:
working example but conflicts with hathi trust module that tries to define same controller

wopsononock committed:
create child directive to avoid collision with HathiTrust and then require prmSearchResultAvailabilityLine scope that browzine depends on in the way it expects

wopsononock committed:
mods to fix layout conflict with Report a Problem link

wopsononock committed:
adding a comment about third iron placeholder directive and cleaning up some empty lines

chryslovelace committed:
attach address form to prm-request-after

chryslovelace committed:
attach address form to prm-request-after

chryslovelace committed:
fix typo

chryslovelace committed:
fix typo

wopsononock commented on 01PITT_INST-ULS_1_5/js/custom.js:
```diff
@@ -267,45 +275,170 @@ angular.module('hathiTrustAvailability', []).constant('hathiTrustBaseUrl', 'http
-      angular.element(document.getElementsByTagName('prm-form-field'))[2].children[0].children[0].style.marginTop='-25px';
+angular
+  .module('addressSelector', [])
+  .constant('addressServiceBaseUrl', 'https://dev-patron.libraries.pitt.edu/')
```
I like how you define the base url in a single place.  I should follow this pattern in the sms branch.  You might be interested in the code Clinton suggested there for detecting the dev or prod site from the primo url.  https://github.com/ulsdevteam/primove-default-view/blob/8b45fee98afe3f6fc39085693bf4e6e7e3f673a3/01PITT_INST-ULS_1_5/js/custom.js#L337

wopsononock commented on 01PITT_INST-ULS_1_5/js/custom.js:
```diff
@@ -267,45 +275,170 @@ angular.module('hathiTrustAvailability', []).constant('hathiTrustBaseUrl', 'http
+    bindings: {parentCtrl: `<`},
+    controller: 'addressSelectorController',
+    template: 
+    '<div layout="row" ng-if="showAddresses()">\
```
Yes! At first glance, this looks like a way cleaner strategy than the watcher-based one I employed.

wopsononock replied:
` eliminates the need for escaping the newlines in ES6

wopsononock commented on 01PITT_INST-ULS_1_5/js/custom.js:
```diff
@@ -267,45 +275,170 @@ angular.module('hathiTrustAvailability', []).constant('hathiTrustBaseUrl', 'http
+    };
+
+    $scope.getAddresses = function() {
+      $scope.statusMessage = 'Loading address information...';
```
there is a functional css-based loading animation in the sms module for use here

wopsononock commented on 01PITT_INST-ULS_1_5/js/custom.js:
```diff
@@ -267,45 +275,170 @@ angular.module('hathiTrustAvailability', []).constant('hathiTrustBaseUrl', 'http
+      let url = addressServiceBaseUrl + 'address/?jwt=' + self.getJwt();
+      $http.delete(url).then(() => {
+        $scope.getAddresses();
+      });
```
error condition?

chryslovelace committed:
cancel button & show 2 lines by default

chryslovelace committed:
cancel button & show 2 lines by default

chryslovelace committed:
pull address when first needed, and some layout

chryslovelace committed:
pull address when first needed, and some layout

chryslovelace committed:
added css loading anim from sms

chryslovelace committed:
added css loading anim from sms

wopsononock committed:
Merge branch 'dev-1.5' into ThirdIron

wopsononock committed:
added address module and changed pdf link text

wopsononock committed:
Merge pull request #19 from ulsdevteam/ThirdIron

Third Iron/Browzine links

alex-wreschnig committed:
Setting up branch 1.5.1. Renamed root directory for Primo VE compatibility, and updated build_zip.

kep31 committed:
Update custom1.css

Added CSS code that Mike created to fix navigation issues caused by ThirdIron Integration.  Also added CSS code to add in a yellow announcement message bar below navigation as well as other code that needed to be adjusted to make this work.

alex-wreschnig committed:
Updating flexbox behavior fix

alex-wreschnig committed:
Setting up 1.6 branch

alex-wreschnig committed:
Bringing ThirdIron API key info in since it's public anyway, to simplify the build process.

alex-wreschnig committed:
Directory move as part of 1.7 branch setup.

alex-wreschnig committed:
Updating build_zip utility as part of 1.7 branch setup.

chryslovelace committed:
wording changes & css class

chryslovelace committed:
Merge branch 'address-form' of https://github.com/ulsdevteam/primove-default-view into address-form

chryslovelace committed:
float form above request form

wopsononock committed:
Added zipcode blacklist and enhanced validation

wopsononock reviewed (changes requested):
Public Services requested zipcode blocklist feature.

wopsononock committed:
only compare 5 digit zips also wording change

wopsononock committed:
minor cleanup

wopsononock committed:
add upstream changes

wopsononock committed:
added final zipcode list
